Rewrite and review

fredrik-krokedil opened the pull request:
In accordance with [ClickUp task 1wgj5f7](https://app.clickup.com/t/1wgj5f7).

Refactored requests, rewrote most of meta-box and went through the rest of the code.

Please review for merge into develop branch.

fredrik-krokedil committed:
Empty request class outline

Contains exactly nothing.

fredrik-krokedil committed:
Basic request structure

The foundation of the reimplemented and improved request structure for KOM

fredrik-krokedil committed:
Main request types implemented

fredrik-krokedil committed:
Some of the specific requests are in

Only request time missing is Refund, as it has a whole bunch of duplicate-looking code I need to look at in more detail.

fredrik-krokedil committed:
Temporary POST Refund class

->get_refund_order_lines() should probably be moved into the WC_Klarna_Order_Management_Order_Lines class to keep things consistent, but having it here will do for now.

fredrik-krokedil committed:
Move actual requests over to new structure

fredrik-krokedil committed:
Accidentally left classes abstract

fredrik-krokedil committed:
Proper Authorization header

fredrik-krokedil committed:
Compatible response decode

Now decodes to an object, like the previous request structure did.

fredrik-krokedil committed:
Update class-kom-request-post-refund.php

TODONE: Implemented refund_reason and refund_amount

fredrik-krokedil committed:
Logger class passed without edit

The logger is already fairly modern and does not need any further work at this time.

fredrik-krokedil committed:
Order lines member type annotation cleanup

fredrik-krokedil committed:
phpcs:ignore clarifications

fredrik-krokedil committed:
Delete class-wc-klarna-order-management-request.php

File is now disused.

fredrik-krokedil committed:
Update class-kom-request-post-refund.php

Note to self: Move refund line functionality to order line handling class.

fredrik-krokedil committed:
Removed surplus debug output

fredrik-krokedil committed:
Meta box cleanup

- Massive readability overhaul of print_standard_content
- New actions and filters for the meta box.
- Option tips shown (and translated!) individually based on what actions are actually selectable.

fredrik-krokedil committed:
Settings class does not need rewrite

Doesn't even smell like pasta. No spaghetti detected.

fredrik-krokedil committed:
Settings page link from plugin list

Just so it matches all the other plugins on this.

fredrik-krokedil committed:
WC_Klarna_Pending_Orders passes without comment

No current need to refactor/rewrite

fredrik-krokedil committed:
WC_Klarna_Sellers_App class passes initial inspection

I see no issues with this class at all.

MichaelBengtsson commented on classes/class-wc-klarna-order-management-order-lines.php:
```diff
@@ -22,42 +22,42 @@ class WC_Klarna_Order_Management_Order_Lines {
 	 * WooCommerce order ID.
 	 *
-	 * @var $order_id
+	 * @var string
```
WooCommerce order IDs are ints generally. Is it a string here?

fredrik-krokedil replied:
The code this was originally called from, in the now removed includes/class-wc-klarna-order-management-request.php, it was declared as a string.
The same value now is in the KOM_Request class, and in trying to keep it as close to the original code as possible I declared it as string there as well.

Tracing it back through the request structure, into WooCommerce and out the other end, yes, it seems to be an int in the very bottom.

I'll update it _everywhere_ to reflect this.

fredrik-krokedil committed:
Removed static keyword from print methods

Strict PHP means they either have to be called differently or be non-static, and as they are the only static functions, I simply made them methods to be in line with the rest of the functions.

MichaelBengtsson commented on classes/request/class-kom-request.php:
```diff
@@ -0,0 +1,347 @@
+	/**
+	 * The WC order id.
+	 *
+	 * @var string
```
As before, these are generally ints. Check if its a string here

MichaelBengtsson reviewed (commented):
Simple stuff to check before QA

fredrik-krokedil committed:
More consistent WC order ID types

fredrik-krokedil committed:
Pass through klarna_order object when capturing

fredrik-krokedil committed:
Strip unused action

Pointed to a now removed method, causing malfunction when no valid actions were found.

NiklasHogefjord commented:
Task linked: [CU-1wgj5f7 Rewrite of Klarna Order management](https://app.clickup.com/t/1wgj5f7)

mntzrr committed:
Remove leading forward slash

mntzrr committed:
Retrieve capture id from HTTP header

mntzrr committed:
Merge branch 'CU-1wgj5f7|Rewrite-of-Klarna-Order-management' into request-rewrite-1wgj5f7
